src/main/java/org/wildfly/security/auth/callback/AuthenticationCompleteCallback.java

```diff
@@ -31,6 +31,10 @@
 
     private static final long serialVersionUID = -8336218311376736914L;
 
+    public static final AuthenticationCompleteCallback SUCCEEDED = new AuthenticationCompleteCallback(true);
+
+    public static final AuthenticationCompleteCallback FAILED = new AuthenticationCompleteCallback(false);
+
```

Comment:
Since these are serializable, there should be a package-private ``Object readResolve()`` method which returns SUCCEEDED or FAILED based on the ``success`` field.